vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java

```diff
@@ -1810,25 +1810,33 @@ private void createClickListener() {
         FormLayout formLayout = new FormLayout();
         Label name = new Label();
         Label age = new Label();
+        Label column = new Label();
 
         // begin-source-example
         // source-example-heading: Item Click Listener
         Grid<Person> grid = new Grid<>();
         grid.setItems(getItems());
-        grid.addColumn(Person::getFirstName).setHeader("First Name");
-        grid.addColumn(Person::getAge).setHeader("Age");
+        grid.addColumn(Person::getFirstName).setHeader("First Name").setKey("First Name").setId("first name");
+        grid.addColumn(Person::getAge).setHeader("Age").setKey("Age");
 
         // Disable selection: will receive only click events instead
         grid.setSelectionMode(Grid.SelectionMode.NONE);
 
         formLayout.add(name, age);
         formLayout.addFormItem(name, "Name");
         formLayout.addFormItem(age, "Age");
+        formLayout.addFormItem(column, "Column");
 
         grid.addItemClickListener(
                 event -> {
                     name.setText(event.getItem().getFirstName());
                     age.setText(String.valueOf(event.getItem().getAge()));
+                    if(event.getColumn().isPresent()) {
+                        column.setText(event.getColumn().get().getKey());
+                    }
+                    else {
+                        column.setText("(undefined column");
+                    }
```

Comment:
1. There shouldn't be a situation where the column is not defined, so the else-part could be just removed.

2. Could be cleaner this way:
`event.getColumn().ifPresent(col -> column.setText(col.getKey()))`
(not blocking)

3. Our code style rules state that there should be a space after `if`. Would be better to fix elsewhere as well, but blocking only for this demo snippet which is shown to users.

Reply:
Done.